tables/cost_optimization_recommendation/cost_optimization_recommendation_table.go

```diff
@@ -0,0 +1,81 @@
+package cost_optimization_recommendation
+
+import (
+	"time"
+
+	"github.com/rs/xid"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const CostOptimizationRecommendationsTableIdentifier = "aws_cost_optimization_recommendation"
+
+// CostOptimizationRecommendationsTable - table for CostOptimizationRecommendations
+type CostOptimizationRecommendationsTable struct{}
+
+// Identifier implements table.Table
+func (t *CostOptimizationRecommendationsTable) Identifier() string {
+	return CostOptimizationRecommendationsTableIdentifier
+}
+
+// GetSourceMetadata implements table.Table
+func (t *CostOptimizationRecommendationsTable) GetSourceMetadata() []*table.SourceMetadata[*CostOptimizationRecommendation] {
+	defaultArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
+		FileLayout: utils.ToStringPointer("%{DATA:export_name}/(?:data/%{DATA:partition}/)?(?:%{INT:from_date}-%{INT:to_date}/)?(?:%{DATA:assembly_id}/)?(?:%{DATA:timestamp}-%{DATA:execution_id}/)?%{DATA:file_name}.csv.(?:zip|gz)"),
```

Comment:
Can the Cost Optimization Recommendations have `%{INT:from_date}-%{INT:to_date}`, or is that pattern mostly found in legacy reports? Similarly, can they have `%{DATA:assembly_id}` and end in `.zip`?

For any file created by AWS Data Export, I thought it'd have to be one of these patterns:
- Create new files: `<bucket-name>/<prefix>/<export-name>/data/<partition>/<timestamp>-<execution-id>/<export-name>-<chunk-number>.csv.gz`
- Overwrite: `<bucket-name>/<prefix>/<export-name>/data/<partition>/<export-name>-<chunk-number>.csv.gz`

Reply:
Seems like the pattern could be simpler if we only accounted for the 2 cases above (unless there are other possible ones not listed in https://docs.aws.amazon.com/cur/latest/userguide/dataexports-export-delivery.html#export-summary)

Reply:
Updated the pattern specific to costoptimization file layout.